source/kernel/src/registry/mod.rs

```diff
@@ -0,0 +1,440 @@
+use core::any::TypeId;
+
+use mnemos_alloc::{containers::HeapFixedVec, heap::HeapGuard};
+use postcard::experimental::max_size::MaxSize;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use uuid::{uuid, Uuid};
+
+use crate::comms::{
+    bbq,
+    kchannel::{KProducer, LeakedKProducer},
+    rosc::Sender,
+};
+
+/// A partial list of known UUIDs of driver services
+pub static KNOWN_UUIDS: &[Uuid] = &[
+    //
+    // Kernel UUIDs
+    //
+
+    // SerialMux
+    uuid!("54c983fa-736f-4223-b90d-c4360a308647"),
+    // Simple Serial Port
+    uuid!("f06aac01-2773-4266-8681-583ffe756554"),
+];
+
+/// A marker trait designating a registerable driver service.
+///
+/// Typically used with [Registry::set] or [Registry::set_konly].
+/// Can typically be retrieved by [Registry::get] or [Registry::get_userspace]
+/// After the service has been registered.
+pub trait RegisteredDriver {
+    /// This is the type of the request sent TO the driver service
+    type Request: 'static;
+
+    /// This is the type of the response sent FROM the driver service
+    type Response: 'static;
+
+    /// This is the UUID of the driver service
+    const UUID: Uuid;
+
+    /// Get the type_id used to make sure that driver instances are correctly typed.
+    /// Corresponds to the same type ID as `(Self::Request, Self::Response)`
+    fn type_id() -> TypeId {
```

Comment:
we might want to make this function return the `TypeId` wrapped in a type without a public constructor, or something, in order to prevent user code from overriding this method?